CONTRIBUTING.md

```diff
@@ -0,0 +1,303 @@
+
+#### Using Cargo
+
+1. [Install `wasm-pack`][install-wasm-pack]
+2. Add the following line to `Cargo.toml` under the `[lib]` heading:
+
+   ```diff
+   [lib]
+   +crate-type = ["cdylib", "rlib"]
+   ```
+
+3. Remove the following line from `Cargo.toml` (dependency on
+   `nickel-lang-utilities`):
+
+   ```diff
+   -nickel-lang-utilities = {path = "utilities", version = "0.1.0"}
+   ```
+
+4. Run `wasm-pack`:
+
+   ```shell
+   wasm-pack build -- --no-default-features --features repl-wasm
+   ```
+
+   A `pkg` directory, containing the corresponding NPM package, should now be
+   available.
+5. (Optional) the generated NPM package is named `nickel`, but this name is not
+   very descriptive, and is already in use in the NPM registry. You can patch
+   the name of the NPM package using `jq` to be `nickel-repl` instead:
+
+   ```shell
+   $ jq '.name = "nickel-repl"' pkg/package.json > package.json.patched \
+     && rm -f pkg/package.json \
+     && mv package.json.patched pkg/package.json
+   ```
+
+   If you don't have `jq`, you can do it by hand: replace the `name` attribute
+   in `pkg/package.json` by `"nickel-repl"`.
+
+The `pkg` directory now contains a `nickel-repl` NPM package that can be used
+from JavaScript.
+
+## Testing
+
+Tests are run via `cargo test`. They are two types of tests:
+
+- Unit tests, located directly in the corresponding module.
+- Integration tests, located in the dedicated crate `tests`.
+
+You can take inspiration from the existing tests to add your own. By convention,
+tests expected to pass are written in a standalone Nickel file in `tests/pass/`.
+Each `.ncl` file defines a list of expressions that must individually evaluate
+to the boolean `true`. The whole file is an expression that returns true if and
+only if every tests pass, or fail with a contract failure to help locating the
+failing test (instead of returning just `false`).
+
+If a test expected to pass is failing, run it directly using nickel with `nickel
+-f tests/pass/test_that_doesnt_pass.ncl` to get better error messages than
+`cargo test`.
+
+Tests expected to fail are often embedded directly into rust source code,
+because you usually want to additionally check that the error is the one you
+expect.
+
+## Benchmarking
+
+If your change is likely to impact performance, it is recommended to run the
+benchmark suite on master and on your branch to assess any performance changes.
+Please report your findings in the description of the PR.
+
+The benchmark suite is located in the `benches/` directory. To run it:
+
+```
+cargo bench
+```
+
+Note that a full run takes some time, up to a dozen of minutes. You can run
+specific benchmarks instead of the full suite. Please refer to the documentation
+of [`cargo bench`][doc-cargo-bench].
+
+# How to submit changes
+
+Once you have made sure the maintainers are aware of your changes, you can start
+the implementation, eventually submitting changes as a pull request.
```

Comment:
This is remarkably condescending. Let's avoid.

Reply:
The idea was to avoid having contributor wasting their time on something that has no chance to land, but maybe this has been already made clear a the beginning. I've removed it.